From: Anthony Sottile <asottile@umich.edu>
Date: Fri, 27 Apr 2018 08:26:38 -0700
Subject: Fix perl imports from . for perl 5.26

---
 Doc/tools/node2label.pl | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Doc/tools/node2label.pl b/Doc/tools/node2label.pl
index 6491b20..44bd434 100755
--- a/Doc/tools/node2label.pl
+++ b/Doc/tools/node2label.pl
@@ -9,7 +9,7 @@ use English;
 $INPLACE_EDIT = '.bak';
 
 # read the labels, then reverse the mappings
-require "labels.pl";
+require "./labels.pl";
 
 %nodes = ();
 my $key;
@@ -29,7 +29,7 @@ foreach $label (sort keys %external_labels) {
 
 # This adds the "internal" labels added for indexing.  These labels will not
 # be used for file names.
-require "intlabels.pl";
+require "./intlabels.pl";
 foreach $label (keys %internal_labels) {
   $key = $internal_labels{$label};
   $key =~ s|^/||;
