From: Felix Krull <f_krull@gmx.de>
Date: Fri, 27 Apr 2018 08:24:08 -0700
Subject: deadsnakes-force-info-docs

---
 Doc/tools/mkinfo | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/Doc/tools/mkinfo b/Doc/tools/mkinfo
index 994fe1d..8351696 100755
--- a/Doc/tools/mkinfo
+++ b/Doc/tools/mkinfo
@@ -64,5 +64,9 @@ sed -i 's/^@top \(.*\)/@top \1 (v2.5)\n@dircategory Python\n@direntry\n* '$DOCID
 
 # generate the .info files:
 
-run $MAKEINFO --footnote-style end --fill-column 72 \
+# We use force here to get texinfo to just run without caring about errors. I
+# think parts of the info documentation have been broken for some time now;
+# there's a syntax error somewhere in the generated PVER-ref.texi. Let's just
+# hope noone uses info anymore and move on.
+run $MAKEINFO --force --footnote-style end --fill-column 72 \
 	      --paragraph-indent 0 --output=$INFONAME $TEXINAME
