From: Felix Krull <f_krull@gmx.de>
Date: Fri, 27 Apr 2018 08:24:07 -0700
Subject: fpectl-module

---
 setup.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/setup.py b/setup.py
index c615642..a438b5d 100644
--- a/setup.py
+++ b/setup.py
@@ -957,6 +957,11 @@ class PyBuildExt(build_ext):
             exts.append( Extension('_curses_panel', ['_curses_panel.c'],
                                    libraries = [panel_library] + curses_libs) )
 
+        # Build `fpectl` module if Python is configured with --with-fpectl
+        data = open('pyconfig.h').read()
+        m = re.search(r"#\s*define\s+WANT_SIGFPE_HANDLER\s+1\s*", data)
+        if m is not None:
+            exts.append(Extension('fpectl', ['fpectlmodule.c']))
 
         # Andrew Kuchling's zlib module.  Note that some versions of zlib
         # 1.1.3 have security problems.  See CERT Advisory CA-2002-07:
