--- a/Lib/linecache.py
+++ b/Lib/linecache.py
@@ -104,7 +104,11 @@
                         )
                         return cache[filename][2]
 
-        # Try looking through the module search path.
+        # Try looking through the module search path, taking care to handle packages.
+
+        if basename == '__init__.py':
+            # filename referes to a package
+            basename = filename
 
         for dirname in sys.path:
             # When using imputil, sys.path may contain things other than
