From: Felix Krull <f_krull@gmx.de>
Date: Fri, 27 Apr 2018 08:24:08 -0700
Subject: readline-6.3-compat

---
 Modules/readline.c | 21 ++++++++++++++++++---
 1 file changed, 18 insertions(+), 3 deletions(-)

diff --git a/Modules/readline.c b/Modules/readline.c
index 2687b53..ba7e078 100644
--- a/Modules/readline.c
+++ b/Modules/readline.c
@@ -712,12 +712,27 @@ setup_readline(void)
 	rl_bind_key_in_map ('\t', rl_complete, emacs_meta_keymap);
 	rl_bind_key_in_map ('\033', rl_complete, emacs_meta_keymap);
 	/* Set our hook functions */
-	rl_startup_hook = (Function *)on_startup_hook;
+	rl_startup_hook =
+#if defined(_RL_FUNCTION_TYPEDEF)
+		(rl_hook_func_t *)on_startup_hook;
+#else
+		(Function *)on_startup_hook;
+#endif
 #ifdef HAVE_RL_PRE_INPUT_HOOK
-	rl_pre_input_hook = (Function *)on_pre_input_hook;
+	rl_pre_input_hook =
+#if defined(_RL_FUNCTION_TYPEDEF)
+		(rl_hook_func_t *)on_pre_input_hook;
+#else
+		(Function *)on_pre_input_hook;
+#endif
 #endif
 	/* Set our completion function */
-	rl_attempted_completion_function = (CPPFunction *)flex_complete;
+	rl_attempted_completion_function =
+#if defined(_RL_FUNCTION_TYPEDEF)
+		(rl_completion_func_t *)flex_complete;
+#else
+		(CPPFunction *)flex_complete;
+#endif
 	/* Set Python word break characters */
 	rl_completer_word_break_characters =
 		strdup(" \t\n`~!@#$%^&*()-=+[{]}\\|;:'\",<>/?");
