From: Felix Krull <f_krull@gmx.de>
Date: Fri, 27 Apr 2018 08:24:05 -0700
Subject: tkinter-import

---
 Lib/lib-tk/Tkinter.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/Lib/lib-tk/Tkinter.py b/Lib/lib-tk/Tkinter.py
index b9c96a3..93f1b7e 100644
--- a/Lib/lib-tk/Tkinter.py
+++ b/Lib/lib-tk/Tkinter.py
@@ -35,7 +35,10 @@ __version__ = "$Revision: 67737 $"
 import sys
 if sys.platform == "win32":
     import FixTk # Attempt to configure Tcl/Tk without requiring PATH
-import _tkinter # If this fails your Python may not be configured for Tk
+try:
+    import _tkinter
+except ImportError, msg:
+    raise ImportError, str(msg) + ', please install the python%s.%s-tk package' % sys.version_info[:2]
 tkinter = _tkinter # b/w compat for export
 TclError = _tkinter.TclError
 from types import *
